pkg/api/types.go

```diff
@@ -2154,3 +2154,38 @@ type RangeAllocation struct {
 	// a single allocated address (the fifth bit on CIDR 10.0.0.0/8 is 10.0.0.4).
 	Data []byte `json:"data"`
 }
+
+// LockSpec is the specification of a resource lock.
+type LockSpec struct {
+	// Who currently has the lock.
+	HeldBy string `json:"heldBy"`
+	// How long the lock is valid.
+	LeaseSeconds uint64 `json:"leaseSeconds"`
+}
+
+type LockStatus struct {
+	// When the lock was first acquired.
+	AcquiredTime util.Time `json:"acquiredTime,omitempty"`
+	// When the lock was last renewed.
+	RenewTime util.Time `json:"renewTime,omitempty"`
```

Comment:
Tense disagrees with AcquiredTime. Also, look at NodeCondition. For times of some most recent occurrence, we use LastFooTime, such as LastTransitionTime. So, this should be LastRenewalTime.